### PR TITLE
Update: added empty string for desc and category check

### PR DESCRIPTION
**Pull request by anikethsaha**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).
- [x] The team has reached consensus on the changes proposed in this pull request. If not, I understand that the evaluation process will begin with this pull request and won't be merged until the team has reached consensus.

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain: internal rule `no-invalid-meta` changes 

#### What changes did you make? (Give an overview)
Added checks for empty `meta.docs.description` and `meta.docs.category` as these should not be empty.

#### Is there anything you'd like reviewers to focus on?
None.

### CODE REVIEW

**Comment by anikethsaha** [comment]
anything to move this forward ?

File: tools/internal-rules/no-invalid-meta.js [hunk 1]
```diff
@@ -161,7 +185,9 @@ module.exports = {
             missingMetaDocsRecommended: "Rule is missing a meta.docs.recommended property.",
             missingMetaSchema: "Rule is missing a meta.schema property.",
             noExport: "Rule does not export anything. Make sure rule exports an object according to new rule format.",
-            incorrectExport: "Rule does not export an Object. Make sure the rule follows the new rule format."
+            incorrectExport: "Rule does not export an Object. Make sure the rule follows the new rule format.",
+            emptyMetaDocsDescription: "Rule's meta.docs.description cant be empty.",
```
Comment (kaicataldo) [inline tools/internal-rules/no-invalid-meta.js:189]:
```suggestion
            emptyMetaDocsDescription: "Rule's meta.docs.description cannot be empty.",
```

File: tools/internal-rules/no-invalid-meta.js [hunk 2]
```diff
@@ -161,7 +185,9 @@ module.exports = {
...
+            emptyMetaDocsCategory: "Rule's meta.docs.category cant be empty."
```
Comment (kaicataldo) [inline tools/internal-rules/no-invalid-meta.js:190]:
```suggestion
            emptyMetaDocsCategory: "Rule's meta.docs.category cannot be empty."
```

**Comment by kaicataldo** [comment]
@eslint/eslint-team Would love some more opinions on this!

**Comment by mdjermanovic** [comment]
Regarding empty `meta.docs.description`, doesn't `eslint-plugin/require-meta-docs-description` already report this?

As for `meta.docs.category`, maybe we could rather make a rule that checks it against a configured list of allowed categories?

File: tools/internal-rules/no-invalid-meta.js [hunk 3]
```diff
@@ -61,6 +61,20 @@ function hasMetaDocsDescription(metaPropertyNode) {
...
 }
 
+
+/**
+ * Checks whether the value of the property is empty or not
+ * @param {ASTNode} metaPropertyNode The `meta` ObjectExpression for this rule.
+ * @param {string}  type the property type to check
+ * @returns {boolean} `true` if propertie valye is empty else `false`
+ */
+function isEmptMetaStringProp(metaPropertyNode, type) {
```
Comment (nzakas) [inline tools/internal-rules/no-invalid-meta.js:71]: Please rename to isEmptyMetaStringProperty. 

(Spelling correction, no need to abbreviate property as we spell it out elsewhere.)

Comment (anikethsaha) [reply tools/internal-rules/no-invalid-meta.js:71]: sure

**Comment by anikethsaha** [comment]
> Regarding empty `meta.docs.description`, doesn't `eslint-plugin/require-meta-docs-description` already report this?

I missed this, yea it does the same. I will remove it from this PR. 


> As for `meta.docs.category`, maybe we could rather make a rule that checks it against a configured list of allowed categories?

I guess we can check the same in this rule itself? in a separate message ?

File: tools/internal-rules/no-invalid-meta.js (hunk 3 above)
Comment (nzakas) [reply tools/internal-rules/no-invalid-meta.js]: Looks like this hasn't been changed yet?

Comment (anikethsaha) [reply tools/internal-rules/no-invalid-meta.js]: did you mean this one https://github.com/eslint/eslint/pull/13179/commits/ba92c63514ced0a517335495784fefac745e7500 ?

Comment (nzakas) [reply tools/internal-rules/no-invalid-meta.js]: "Empty" is still spelled incorrectly (missing the "y')

**Comment by mdjermanovic** [comment]
> > As for `meta.docs.category`, maybe we could rather make a rule that checks it against a configured list of allowed categories?
> 
> I guess we can check the same in this rule itself? in a separate message?

I'm not sure. Since it's an internal rule we could do this as a part of `no-invalid-meta` to avoid duplicating some code, but I guess ideally it would be a configurable rule for itself. Another opinion would be appreciated!

Either way, this looks like a useful check. It seems that a typo or a wrong uppercase/lowercase in rule's category may be committed and go unnoticed until release time. Then, I believe, the `gensite` script would throw [here](https://github.com/eslint/eslint/blob/748734fdd497fbf61f3a616ff4a09169135b9396/Makefile.js#L205).

**Comment by mdjermanovic** [comment]
> As for `meta.docs.category`, maybe we could rather make a rule that checks it against a configured list of allowed categories?

There's actually an accepted issue for this rule in `eslint-plugin-eslint-plugin`:

not-an-aardvark/eslint-plugin-eslint-plugin#60

File: tools/internal-rules/no-invalid-meta.js (hunk 3 above)
Comment (anikethsaha) [reply tools/internal-rules/no-invalid-meta.js]: done

**Comment by anikethsaha** [comment]
there is a PR which is open for the issue mentioned above, https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/pull/63 . 

I believe that would fix the category check implemented here. So if that PR is being merged and shipped. then I guess this PR is good to close. As description check is already implemented and for category check, there is an open PR in `eslint-plugin-eslint-plugin`

**Comment by mdjermanovic** [comment]
I'd agree to close this, I think it's too much code to check only one invalid case. A rule with a list of valid categories would be nice (either in the plugin or as an internal rule) if we don't decide to just remove `meta.docs.category` in regard to #13398.

**Comment by anikethsaha** [comment]
> I'd agree to close this, I think it's too much code to check only one invalid case. A rule with a list of valid categories would be nice (either in the plugin or as an internal rule) if we don't decide to just remove `meta.docs.category` in regard to #13398.

yea, I have similar thoughts
If we are going to remove `meta.docs.category` , then it doesnt make sense to land this change.

**Comment by nzakas** [comment]
Sounds like there's general agreement to close this without merging. I agree, so closing.

**Pull request closed without merging**